src/java/org/apache/cassandra/index/Index.java

```diff
@@ -424,7 +465,8 @@ public Indexer indexerFor(DecoratedKey key,
                               RegularAndStaticColumns columns,
                               int nowInSec,
                               WriteContext ctx,
-                              IndexTransaction.Type transactionType);
+                              IndexTransaction.Type transactionType,
+                              Memtable memtable);
```

Comment:
This breaks index API compatibility without a default implementation, which is fine if this is going to ship in a major.

Reply:
Yup. Something we can revisit once more of the pieces of SAI are in place and we have full visibility on how difficult it would be to make things backwards compatible overall.